src/govuk/components/checkboxes/checkboxes.js

```diff
@@ -29,13 +30,32 @@ Checkboxes.prototype.init = function () {
     // If we have content that is controlled, set attributes.
     $input.setAttribute('aria-controls', controls)
     $input.removeAttribute('data-aria-controls')
-    this.setAttributes($input)
-  }.bind(this))
+  })
+
+  if (document.readyState === 'loading') {
+    window.addEventListener('DOMContentLoaded', this.syncState.bind(this))
+  } else {
+    this.syncState()
+  }
+
+  // When the page is restored after navigating 'back' in some browsers the
+  // state of form controls is not restored until *after* the DOMContentLoaded
+  // event is fired, so we need to sync after the pageshow event is fired as
+  // well.
+  //
+  // (Older browsers don't have a pageshow event, so we do both.)
+  window.addEventListener('pageshow', this.syncState.bind(this))
 
   // Handle events
   $module.addEventListener('click', this.handleClick.bind(this))
 }
 
+Checkboxes.prototype.syncState = function () {
+  nodeListForEach(this.$inputs, function ($input) {
```

Comment:
I can't unsubscribe from all notifications so sometimes I see interesting things, excellent write up, super interesting...

You might be able to simplify this to:

`nodeListForEach(this.$inputs, this.setAttributes($input).bind(this))`

Reply:
Thanks, @nickcolley! 👋

Reply:
I think it'd have to be:

```js
nodeListForEach(this.$inputs, this.setAttributes.bind(this))
```

which is possibly a little less clear – any thoughts @hannalaakso?